api/playlist/playlist.controller.js

```diff
@@ -0,0 +1,53 @@
+const { handleError } = require('../../utils');
+const Playlist = require('./playlist.model');
+
+const controller = {};
+
+controller.lists = (req, res) => {
+  Playlist.find({}).exec()
+    .then(playlists => res.send(playlists))
+    .catch(handleError(res));
+};
+
+controller.show = (req, res) => {
+  const id = req.params.playlistId;
+  Playlist.findOne({ id }).exec()
+    .then(playlist => res.send(playlist))
+    .catch(handleError(res));
+};
+
+controller.create = (req, res) => {
+  const { name, owner } = req.body;
+  Playlist.create({ name, owner })
+    .then(playlist => res.send(playlist))
+    .catch(handleError(res));
+};
+
+controller.delete = (req, res) => {
+  Playlist.remove({ _id: req.params.playlistId })
+    .then(response => res.send(response))
+    .catch(handleError(res));
+};
+
+controller.addPodcast = (req, res) => {
+  const { playlistId, podcastId } = req.params;
+  Playlist.findOne({ _id: playlistId }).exec()
+    .then(list => {
+      list.podcasts.push(podcastId);
+      list.save();
+    })
+    .then(() => res.send('success'))
+    .catch(handleError(res));
+};
+
+controller.removePodcast = (req, res) => {
+  const { playlistId, podcastId } = req.params;
+  Playlist.update(
+    { _id: playlistId },
+    { $pull: { podcasts: podcastId } }
+    ).exec()
+    .then(res.send('successfully removed'))
```

Comment:
I think we want to send a different status code. Look up what is usually sent when deleting